cpmonitor/management/commands/cleanupimages.py

```diff
@@ -0,0 +1,212 @@
+
+        if is_dry_run:
+            self.stdout.write(
+                self.style.SUCCESS(
+                    "Dry run: '%d' images will be cleaned up in directory '%s': '%s'"
+                    % (
+                        len(orphan_images),
+                        os.path.abspath(IMAGES_PATH),
+                        orphan_images,
+                    )
+                )
+            )
+
+            return
+
+        for orphan_image in orphan_images:
+            os.remove(orphan_image)
+
+        if len(orphan_images):
+            self.stdout.write(
+                self.style.SUCCESS(
+                    "Cleaned up '%d' images in directory '%s': '%s'"
+                    % (
+                        len(orphan_images),
+                        os.path.abspath(IMAGES_PATH),
+                        orphan_images,
+                    )
+                )
+            )
+        else:
+            self.stdout.write(self.style.SUCCESS("No images found to clean up"))
+
+    def resolve_orphan_images(self) -> list[str]:
+        referenced_images = self._fetch_referenced_images()
+        saved_images = self.resolve_saved_images()
+
+        orphan_images = []
+        for saved_image in saved_images:
+            if saved_image not in referenced_images:
+                orphan_images.append(saved_image)
+
+        return orphan_images
+
+    def _fetch_referenced_images(self) -> list[str]:
+        text_field_images = self._resolve_image_paths_from_text_fields()
+        file_field_images = self._resolve_image_paths_from_file_fields()
+        distinct_images = list(dict.fromkeys(text_field_images + file_field_images))
+
+        return distinct_images
+
+    def _resolve_image_paths_from_text_fields(self) -> list[str]:
+        relative_paths = []
+
+        for model_class in apps.get_models():
+            models = model_class.objects.all()
+
+            for field in model_class._meta.fields:
+                if self._is_domain_text_field(field):
+                    relative_paths += self._extract_file_paths_from_models(
+                        field, models
+                    )
+
+        absolute_paths = []
+        base_path = os.path.abspath(BASE_PATH)
+
+        for relative_path in relative_paths:
+            absolute_paths.append(os.path.join(base_path, relative_path.strip("/")))
+
+        return absolute_paths
+
+    def _is_domain_text_field(self, field) -> bool:
+        return self._is_domain_model(field) and isinstance(
+            field, DjangoModels.TextField
+        )
+
+    def _is_domain_model(self, field) -> bool:
+        return field.model._meta.app_label == APP_NAME
+
+    def _extract_file_paths_from_models(self, field, models) -> list[str]:
+        file_paths = []
+        field_values = self._extract_values_from_models(field, models)
+
+        for field_value in field_values:
+            if self._contains_text_field_file_path(field, field_value):
```

Comment:
Why don't you directly extract them, now you do the same thing twice :thinking:

Reply:
If I would extract them directly this would lead to not so clean code (in my opinion) because I have to check if the file_path should be added to file_paths list (and a future developer could ask why there is such a check?). Due to this is not a performance critical process I think it's ok to do the same thing twice when it leads to more readable code. What do you think?

Reply:
I understand that we like to do the `_is_domain_text_field` check beforehand, but for the file paths we anyway check if they can be added: 
```
  for file_path in potential_file_paths:
            if not self._is_url(file_path):
                real_file_paths.append(file_path)
```
But maybe I am missing something here...

Reply:
Yes but there is a difference:

- _contains_text_field_file_path() ensures that it is a TEXT field that contains a file path
- _is_url() ensures only for a TEXT field that it is a really a file path and not only an url to an image (which will not be saved on the file system)
- _contains_file_field_file_path() ensures that it is a FILE field that contains a file path (no url possible)

Or do I understand you wrong 😄

Reply:
Adapted as discussed